core/mm/core_mmu.c

```diff
@@ -628,9 +628,18 @@ static void verify_special_mem_areas(struct tee_mmap_region *mem_map,
 	/*
 	 * Check memories do not intersect any mapped memory.
 	 * This is called before reserved VA space is loaded in mem_map.
+	 *
+	 * Exceptions are the memory areas that maps with the same attributes
+	 * as for example MEM_AREA_RAM_NSEC and MEM_AREA_NSEC_SHM
+	 * which may overlap since they are used for the same purpose
+	 * except that MEM_AREA_NSEC_SHM is always mapped and
+	 * MEM_AREA_RAM_NSEC only uses a dynamic mapping.
 	 */
 	for (mem = start; mem < end; mem++) {
 		for (mmap = mem_map; mmap->type != MEM_AREA_END; mmap++) {
+			if (core_mmu_type_to_attr(mem->type) ==
```

Comment:
I'd rather be explicit about which memory area we allow to overlap. The purpose here is to check against unintended overlapping, so let's make it easy to keep track of the allowed exceptions.
Currently, it seems it's only with `MEM_AREA_RAM_NSEC` and `MEM_AREA_NSEC_SHM` where it makes sense to allow an overlap.

Reply:
fair. i'll update accordingly.

Reply:
I've remove that ch